core/base/src/TSystem.cxx

```diff
@@ -232,6 +232,9 @@ Bool_t TSystem::Init()
 
 void TSystem::SetProgname(const char *name)
 {
+   if (gProgName)
```

Comment:
What about 'just' ```delete [] gProgName;```?

Reply:
I can do it here, but I just open large #4874, where I can improve this as well